src/GovUk.Education.ExploreEducationStatistics.Data.Services/PublicationMetaService.cs

```diff
@@ -1,40 +1,70 @@
 using AutoMapper;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
 using GovUk.Education.ExploreEducationStatistics.Data.Model;
 using GovUk.Education.ExploreEducationStatistics.Data.Model.Services.Interfaces;
 using GovUk.Education.ExploreEducationStatistics.Data.Services.Interfaces;
 using GovUk.Education.ExploreEducationStatistics.Data.Services.ViewModels.Meta;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Cosmos.Table;
+using static GovUk.Education.ExploreEducationStatistics.Common.TableStorageTableNames;
 
 namespace GovUk.Education.ExploreEducationStatistics.Data.Services
 {
     public class PublicationMetaService : IPublicationMetaService
     {
         private readonly IReleaseService _releaseService;
         private readonly ISubjectService _subjectService;
+        private readonly ITableStorageService _tableStorageService;
         private readonly IMapper _mapper;
 
-        public PublicationMetaService(
-            IReleaseService releaseService,
+        public PublicationMetaService(IReleaseService releaseService,
             ISubjectService subjectService,
+            ITableStorageService tableStorageService,
             IMapper mapper)
         {
             _releaseService = releaseService;
             _subjectService = subjectService;
+            _tableStorageService = tableStorageService;
             _mapper = mapper;
         }
 
-        public PublicationSubjectsMetaViewModel GetSubjectsForLatestRelease(Guid publicationId)
+        public async Task<Either<ActionResult, PublicationSubjectsMetaViewModel>> GetSubjectsForLatestRelease(
+            Guid publicationId)
         {
             var releaseId = _releaseService.GetLatestPublishedRelease(publicationId);
-            var subjects = _subjectService.GetSubjectsForReleaseAsync(releaseId.Value).Result;
-            var subjectMetaViewModels = _mapper.Map<IEnumerable<IdLabel>>(subjects);
+            if (!releaseId.HasValue)
+            {
+                return new NotFoundResult();
+            }
 
             return new PublicationSubjectsMetaViewModel
             {
                 PublicationId = publicationId,
-                Subjects = subjectMetaViewModels
+                Highlights = await GetHighlights(releaseId.Value),
+                Subjects = await GetSubjects(releaseId.Value)
             };
         }
+
+        private async Task<IEnumerable<IdLabel>> GetHighlights(Guid releaseId)
+        {
+            var filter = TableQuery.GenerateFilterCondition(nameof(ReleaseFastTrack.PartitionKey),
+                QueryComparisons.Equal, releaseId.ToString());
+            var query = new TableQuery<ReleaseFastTrack>().Where(filter);
+
+            return (await _tableStorageService.ExecuteQueryAsync(PublicReleaseFastTrackTableName, query))
+                .Where(releaseFastTrack => !string.IsNullOrEmpty(releaseFastTrack.HighlightName))
```

Comment:
Rather than grabbing every fast track for a release, could we use a combined filter for the query that would translate to `PartitionKey = releaseId AND HighlightName != '' AND HighlightName != null`?

If the nullability is a problem, I would suggest making sure that `HighlightName` is always a string, even if it's an empty one.

Reply:
Thanks, just made this change.
Table rows that have a null HighlightName don't have that property at all.

Adding a filter to say HighlightName property != '' is also restricting results to those which have HighlightName so in effect `HighlightName != '' AND HighlightName != null`

Reply:
Cool, nice one :+1: